src/MessageHandler/WarmUpGroupedUserCacheMessageHandler.php

```diff
@@ -0,0 +1,146 @@
+ */
+
+declare(strict_types=1);
+
+namespace OAT\SimpleRoster\MessageHandler;
+
+use Doctrine\Common\Cache\CacheProvider;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\EntityNotFoundException;
+use Doctrine\ORM\NonUniqueResultException;
+use OAT\SimpleRoster\Exception\CacheWarmupException;
+use OAT\SimpleRoster\Exception\DoctrineResultCacheImplementationNotFoundException;
+use OAT\SimpleRoster\Generator\UserCacheIdGenerator;
+use OAT\SimpleRoster\Message\WarmUpGroupedUserCacheMessage;
+use OAT\SimpleRoster\Repository\UserRepository;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Messenger\Handler\MessageHandlerInterface;
+use Throwable;
+
+class WarmUpGroupedUserCacheMessageHandler implements MessageHandlerInterface
+{
+    /** @var EntityManagerInterface */
+    private $entityManager;
+
+    /** @var UserRepository */
+    private $userRepository;
+
+    /** @var UserCacheIdGenerator */
+    private $cacheIdGenerator;
+
+    /** @var CacheProvider */
+    private $resultCacheImplementation;
+
+    /** @var LoggerInterface */
+    private $messengerLogger;
+
+    /** @var LoggerInterface */
+    private $cacheWarmupLogger;
+
+    /** @var int */
+    private $cacheTtl;
+
+    /**
+     * @throws DoctrineResultCacheImplementationNotFoundException
+     */
+    public function __construct(
+        EntityManagerInterface $entityManager,
+        UserRepository $userRepository,
+        UserCacheIdGenerator $cacheIdGenerator,
+        LoggerInterface $messengerLogger,
+        LoggerInterface $cacheWarmupLogger,
+        int $userWithAssignmentsCacheTtl
+    ) {
+        $this->entityManager = $entityManager;
+        $this->userRepository = $userRepository;
+        $this->cacheIdGenerator = $cacheIdGenerator;
+        $this->messengerLogger = $messengerLogger;
+        $this->cacheWarmupLogger = $cacheWarmupLogger;
+        $this->cacheTtl = $userWithAssignmentsCacheTtl;
+
+        $resultCacheImplementation = $entityManager->getConfiguration()->getResultCacheImpl();
+
+        if (!$resultCacheImplementation instanceof CacheProvider) {
+            throw new DoctrineResultCacheImplementationNotFoundException(
+                'Doctrine result cache implementation is not configured.'
+            );
+        }
+
+        $this->resultCacheImplementation = $resultCacheImplementation;
+    }
+
+    /**
+     * @throws Throwable
+     */
+    public function __invoke(WarmUpGroupedUserCacheMessage $message): void
+    {
+        foreach ($message->getUsernames() as $username) {
+            try {
+                $this->refreshCacheForUsername($username);
+            } catch (Throwable $exception) {
+                $errorLog = sprintf(
+                    "Unsuccessful cache warmup for user '%s'. Error: %s",
+                    $username,
+                    $exception->getMessage()
+                );
+
+                $this->messengerLogger->error($errorLog);
+                $this->cacheWarmupLogger->error($errorLog);
```

Comment:
Are you sure we want to replicate the same log into different channels?

Reply:
Fixed based on our discussion